### PR TITLE
GAWB-2811 Handle SA already existing on Google

### PR DESCRIPTION
**Pull request by helgridly**
This happens when you start a FiaB and ask Sam for a pet who already exists in the Google project but doesn't exist in its local LDAP.

Friends with https://github.com/broadinstitute/workbench-libs/pull/35

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ object Dependencies {
 
   val workbenchUtilV   = "0.2-72adc94"
   val workbenchModelV  = "0.4-72adc94"
-  val workbenchGoogleV = "0.5-72adc94"
+  val workbenchGoogleV = "0.5-58c3e7b-SNAP"
```
Comment (rtitle) [inline project/Dependencies.scala]: Just noting when https://github.com/broadinstitute/workbench-libs/pull/36 is picked up the `workbenchGoogle` dependency will need to change to something like `... % "test->test;compile->compile"`.

**Review by rtitle — APPROVED** [review]
Thanks for fixing this

File: project/Dependencies.scala (hunk 1 above)
Comment (helgridly) [reply project/Dependencies.scala]: I tinkered with Leo until it worked and found that `"org.broadinstitute.dsde.workbench" %% "workbench-google"  % workbenchGoogleV % "test" classifier "tests"` worked. `test->test;compile->compile` didn't, I don't think

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala [hunk 2]
```diff
@@ -104,8 +104,9 @@ class UserService(val directoryDAO: DirectoryDAO, val googleDirectoryDAO: Google
     directoryDAO.getPetServiceAccountForUser(user.id).flatMap {
       case Some(email) => Future.successful(email)
       case None =>
-        // First create the service account in Google, which generates a unique id and email
-        googleIamDAO.createServiceAccount(petServiceAccountConfig.googleProject, petSa, petSaDisplayName).flatMap { petServiceAccount =>
+        // First find or create the service account in Google, which generates a unique id and email
+        val petSA = googleIamDAO.getOrCreateServiceAccount(petServiceAccountConfig.googleProject, petSa, petSaDisplayName)
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala]: Rename the `petSa` defined on line 102 to reflect that it's the SA ID

**Review by jmthibault79 — APPROVED** [review]

**Review by MatthewBemis — APPROVED** [review]

**Pull request merged**